Assignment4/AlexRouge - CA43899MT/task07.py

```diff
@@ -0,0 +1,144 @@
+# -*- coding: utf-8 -*-
+"""Task07.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1eS7bihOCI7PBV-u5OucIKaK8KYzwgBq8
+
+**Task 07: Querying RDF(s)**
+"""
+
+!pip install rdflib
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2023-2024/master/Assignment4/course_materials"
+
+"""First let's read the RDF file"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
+"""**TASK 7.1: List all subclasses of "LivingThing" with RDFLib and SPARQL**"""
+
+from rdflib.plugins.sparql import prepareQuery
+
+ns = Namespace("http://somewhere#")
+
+q1 = prepareQuery('''
+  SELECT ?Subclass WHERE {
+    ?Subclass rdfs:subClassOf/rdfs:subClassOf* ns:LivingThing.
+  }
+  ''',
+  initNs = { "rdfs":RDFS, "ns":ns}
+)
+# Visualize the results
+print("RDFLib")
+for s, p, o in g.triples((None, RDFS.subClassOf, ns.LivingThing)):
```

Comment:
This does not provide the same results as the previous sparql function. Use a recursive function